runtime-sdk/src/modules/accounts/mod.rs

```diff
@@ -459,6 +459,10 @@ impl API for Module {
     }
 
     fn mint<C: Context>(ctx: &mut C, to: Address, amount: &token::BaseUnits) -> Result<(), Error> {
+        if amount.amount() == 0 {
```

Comment:
Any reason why not adding the same check to `transfer`? Also I noticed that only `transfer` has the `is_check_only` check.